### ISSUE TITLE
"Scrollable region must have keyboard access" with focusable elements within

### ISSUE THREAD

**Issue by gportela85** [dequelabs/axe-core#4830]
### Product

axe-core

### Product Version

4.10.3

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
No warnings about scrollable regions needing keyboard access.

#### How to Reproduce
Open https://plnkr.co/edit/IyLmkiUknP2hB8jT?open=main.js&preview
Click the arrows at the top right to show the example in a separate window
Run the Axe Extension to scan the page.

#### Additional context
This grid is wrapped with a "tab guard" element (tabIndex 0) that manages which cell needs to be focused when the grid is focused, and then the cells (tabIndex -1) will scroll the grid with a custom scroll logic while moving focus across cells.

**Comment by WilcoFiers** [comment]
@YuriScarbaci I would not think spinbutton is appropriate for that kind of a control. I get the idea, since you are "spinning" the carrousel, but spin buttons are intended for changing a form value up and down, not for controlling a user interface. Whether that'd qualify as a WCAG failure I'm not sure though. That feels like an edge case to me. We'll keep it in mind when we pick this up.

**Comment by YuriScarbaci** [comment]
just a question regarding this kind of issues,

Does the axe-core team prefer to have false-positives or false-negatives?

This may sound like a dumb question but if such scenarios are "edge cases" knowing what axe-core teams prefer to address does shape how CI&CD checks should be treated moving forward

I personally would love to be able to configure it somehow so rules that are kinda difficult to check for like the scollable region one can be toggled accordingly until proper fixes comes from you
